src/DocLayer.actor.cpp

```diff
@@ -875,6 +882,13 @@ int main(int argc, char** argv) {
 			}
 			break;
 		}
+		case OPT_FDB_DC_ID: {
+			const char* fdbDatacenterIdStr = args.OptionArg();
+			if (fdbDatacenterIdStr) {
+				fdbDatacenterid = Optional<StringRef>(LiteralStringRef(fdbDatacenterIdStr));
```

Comment:
`LiteralStringRef` is designed to use with constant strings `const char *`. It calls `sizeof()` on the string, that would be quite big. Best way to convert from `std::string` to `StringRef` is `StringRef(s)`. Also by passing `StringRef` we have to worry about destruction of string, which is fine here as it's in `main` which lives forever. I think here best way is to just pass `std:string` all the way. And convert to `StringRef` while calling `setOption`.

Reply:
Sure thing.

Reply:
Not sure what I was reading. Ignore my comment `std::string`. But, please just pass around "const char*` instead `StringRef`

Reply:
Done  67c9969